src/types.jl

```diff
@@ -735,6 +735,29 @@ end
 
 Base.isequal(a::Mul, b::Mul) = isequal(a.coeff, b.coeff) && isequal(a.dict, b.dict)
 
+Base.isequal(b::Add, a::Mul) = isequal(a, b)
+function Base.isequal(a::Mul, b::Add) # this shows up a lot when merging dicts
```

Comment:
Maybe this should just be `false`? I think we cannot construct `1 * x^1 * y^1`. @shashi

Reply:
yes,